File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand_a, strand_b)
    raise ArgumentError unless strand_a.length == strand_b.length
    pairs = strand_a.chars.zip(strand_b.chars)
    pairs.count { |(first, second)| first != second }
  end
end

module BookKeeping
  VERSION = 3
end

File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {
    fun compute(leftStrand: String, rightStrand: String): Int {
        if(leftStrand.count() != rightStrand.count()) throw IllegalArgumentException("leftStrand and rightStrand must be of equal length.")
        
        return leftStrand
                .zip(rightStrand)
                .sumBy { if(it.first == it.second) 0 else 1 }
    }
}